fetch.bs

```diff
@@ -3342,86 +3345,149 @@ steps:
+
+    <li><p>If both the <var>revalidation</var> and <var>partial</var> flags are unset, set
+    <var>response</var> to <var>storedResponse</var>.
+  </ol>
 
  <li>
+  <!-- If response is still null, we require a forward request. -->
   <p>If <var>response</var> is null, run these substeps:
 
   <ol>
    <li><p>If <var>httpRequest</var>'s <a for=request>cache mode</a> is
-   "<code>only-if-cached</code>", then return a
-   <a>network error</a>.
+   "<code>only-if-cached</code>", then return a <a>network error</a>.
 
-   <li><p>Set <var>response</var> to the result of making an
-   <a>HTTP-network fetch</a> using <var>httpRequest</var>
-   with <i>credentials flag</i> if set.
-  </ol>
+   <li><p>Set <var>forwardResponse</var> to the result of making an <a>HTTP-network fetch</a> using
+   <var>httpRequest</var> with <i>credentials flag</i> if set.
 
- <li>
-  <p>If <var>response</var>'s <a for=response>status</a> is <code>304</code> and
-  <var>httpRequest</var>'s <a for=request>cache mode</a> is either
-  "<code>default</code>" or "<code>no-cache</code>", run these substeps:
+   <li><p>If <var>httpRequest</var>'s <var>method</var> is unsafe and <var>forwardResponse</var>'s
+   <a for=response>status</a> begins with <code>2</code> or <code>3</code>, invalidate appropriate
+   stored responses in the HTTP cache, as per the 
+   "<a href=https://tools.ietf.org/html/rfc7234#section-4.4>Invalidation</a>" chapter of 
+   <cite>HTTP Caching</cite> [[!HTTP-CACHING]], and set <var>storedResponse</var> to null.
 
-  <ol>
-   <li><p>Set <var>cachedResponse</var> to the result of selecting a stored
-   <a for=/>response</a> from the HTTP cache using <var>httpRequest</var>, as
-   per the
-   "<a href=https://tools.ietf.org/html/rfc7234#section-4.3.4>Freshening Stored Responses upon Validation</a>"
-   chapter of <cite>HTTP Caching</cite>. [[!HTTP-CACHING]]
+   <li>
+    <p>If the <var>revalidating</var> flag is set and <var>forwardResponse</var>'s 
+    <a for=response>status</a> is <code>304</code>, run these substeps:
 
-   <li><p>If <var>cachedResponse</var> is null (i.e., one cannot be selected), return a
-   <a>network error</a>.
+    <ol>
+     <li><p>Update <var>storedResponse</var>'s <a for=response>header list</a> using
+     <var>forwardResponse</var>'s <a for=response>header list</a>, as per the 
+     "<a href=https://tools.ietf.org/html/rfc7234#section-4.3.4>Freshening Stored Responses upon Validation</a>"
+     chapter of <cite>HTTP Caching</cite> [[!HTTP-CACHING]].
 
-   <li><p>Update <var>cachedResponse</var>'s <a for=response>header list</a>
-   using <var>response</var>'s <a for=response>header list</a>, as per the
-   "<a href=https://tools.ietf.org/html/rfc7234#section-4.3.4>Freshening Stored Responses upon Validation</a>"
-   chapter of <cite>HTTP Caching</cite>. [[!HTTP-CACHING]]
+     <li><p>Set <var>response</var> to <var>storedResponse</var>.
+
+     <p class="note">This updates the stored response in cache as well.
```

Comment:
It seems this note is more fitting for the paragraph above? Also, the formatting is a little off. If you have multiple `<p>` descendants, they need to be on their own line and indented by one.